### ISSUE TITLE
flycheck don't work

### ISSUE THREAD

**Issue by francesco-cadei** [ananthakumaran/tide#451]
## Checklist

- [x] I have searched both open and closed issues and cannot find a duplicate.
- [ ] I can [reproduce the problem](https://github.com/ananthakumaran/tide/wiki/Explanations-about-the-Issue-Reporting-Checklist#use-the-latest-version-of-the-relevant-packages) with the latest version of the relevant packages.
- [x] The problem still occurs after I issued [`M-x tide-restart-server`](https://github.com/ananthakumaran/tide/wiki/Explanations-about-the-Issue-Reporting-Checklist#try-m-x-tide-restart-server) in the buffer where I had the problem.
- [x] I verified that the version and the configuration file path reported by [`M-x tide-verify-setup`](https://github.com/ananthakumaran/tide/wiki/Explanations-about-the-Issue-Reporting-Checklist#check-the-information-reported-by-m-x-tide-verify-setup) are correct.
- [ ] If [tide *is reporting*](https://github.com/ananthakumaran/tide/wiki/Explanations-about-the-Issue-Reporting-Checklist#the-behavior-of-cli-tools-agrees-with-your-expectations) an error or warning I think *should not be reported*, I can run `tsc` (and `tslint`, if applicable) without the error or warning I'm seeing in tide.
- [ ] If [tide is *not reporting*](https://github.com/ananthakumaran/tide/wiki/Explanations-about-the-Issue-Reporting-Checklist#the-behavior-of-cli-tools-agrees-with-your-expectations) an error or warning I think *should be reported*, `tsc` (or `tslint`, if applicable) reports the error or warning I was expecting to see.
- [ ] I am [positive](https://github.com/ananthakumaran/tide/wiki/Explanations-about-the-Issue-Reporting-Checklist#how-to-determine-which-project-the-problem-belongs-to) the problem does not belong to `typescript-mode` or `tsserver`.

## Relevant Version Numbers

- Tide: 4.5.4
- TypeScript: 4.7.4
- Emacs: 27.1

## Steps to Reproduce the Bug

emacs settings:
```emacs-lisp
(use-package flycheck
  :config
  (setq-default flycheck-disabled-checkers '(emacs-lisp-checkdoc))
  (global-flycheck-mode))

(use-package rjsx-mode
  :mode ("\\.js\\'" . rjsx-mode))

(use-package tide
  :config
  (defun setup-tide-mode ()
    "Setup function for tide."
    (interactive)
    (tide-setup)
    (setq flycheck-check-syntax-automatically '(save mode-enabled))
    (eldoc-mode)
    (tide-hl-identifier-mode))
  :hook (rjsx-mode . setup-tide-mode))
```
test.js (example)
```javascript
function createQuote(quote, callback){
    var myQuote = "Like I always say, " + quote;
    callback(myQuote);
}

function logQuote(quote){
    console.log(quote);
}

createQuote("eat your vegetables!", logQuote);
```
jsconfig.json (example)
```json
{
    "compilerOptions": {
	"module": "commonjs",
	"target": "es6"
    }
}
```

errors are colored or underlined yellow or red and if you pass over it with the cursor, the error is explained but:
- there is no arrows on the left indicating the errors 
- `C-c ! l` do nothing

**Comment by ananthakumaran** [comment]
Thanks for the detailed bug report. This is because there is no predefined flycheck checker for this combination (we do support rjsx-mode for jsx files), though it's straightforward to define one

```elisp
(use-package tide
  :ensure t
  :config
  (flycheck-define-generic-checker 'rjs-tide
    "A JS syntax checker using tsserver."
    :start #'tide-flycheck-start
    :verify #'tide-flycheck-verify
    :modes '(rjsx-mode)
    :predicate (lambda ()
                 (and
                  (tide-file-extension-p "js")
                  (tide-flycheck-predicate))))

  (add-to-list 'flycheck-checkers 'rjs-tide t)

  (defun setup-tide-mode ()
    "Setup function for tide."
    (interactive)
    (tide-setup)
    (flycheck-mode +1)
    (setq flycheck-check-syntax-automatically '(save mode-enabled))
    (eldoc-mode +1)
    (tide-hl-identifier-mode))
  :hook (rjsx-mode . setup-tide-mode))
```

**Comment by francesco-cadei** [comment]
Thanks for the support. 
Unfortunately this does not solve the problem, I tried to use only `js2-mode` but nothing even like that.
The problem also exists with `.ts` files, the only difference between `.js` and `.ts` is that in the former errors are colored yellow.

**Comment by ananthakumaran** [comment]
![image](https://user-images.githubusercontent.com/149238/186207971-9299ec21-3887-4c2f-a35c-2a58d1a925b9.png)

I tried to reproduce the issue with [emacs sandbox](https://github.com/alphapapa/with-emacs.sh) with the config you shared plus the flycheck checker definition I shared in the above comment. I have no issue with viewing error list.

> errors are colored or underlined yellow or red and if you pass over it with the cursor, the error is explained

Since you see all the details except the list, my guess is, you have issues unrelated to tide.
